samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnClusterResourceManager.java

```diff
@@ -202,7 +203,9 @@ public void requestResources(SamzaResourceRequest resourceRequest) {
               capability,
               new String[]{preferredHost},
               null,
-              priority);
+              priority,
+              true,
+              label);
```

Comment:
Old constructor looks like:
```java
public ContainerRequest(Resource capability, String[] nodes, 
      String[] racks, Priority priority) {
        this(capability, nodes, racks, priority, true, null);
}
```
As you can see here default value for `relaxLocality` is `true` and for `nodeLabelsExpression` is `null`. So, it shouldn't break any compatibility.